pipelines/build/common/config_regeneration.groovy

```diff
@@ -0,0 +1,433 @@
+
+    return create
+  }
+
+  /**
+  * Queries the Jenkins API. Used to get the pipeline and downstream job details.
+  * @param query
+  */
+  def queryJenkinsAPI(String query) {
+    try {
+      def parser = new JsonSlurper()
+      def get = new URL(query).openConnection()
+      def response = parser.parseText(get.getInputStream().getText())
+      return response
+
+    } catch (Exception e) {
+      // Failed to connect to jenkins api or a parsing error occured
+      context.println "[ERROR] Failure on jenkins api connection or parsing.\nError: ${e}"
+      currentBuild.result = "FAILURE"
+    } 
+  }
+
+  /**
+  * Main function. Ran from regeneration_pipeline.groovy, this will be what jenkins will run first. 
+  */ 
+  @SuppressWarnings("unused")
+  def regenerate() {
+    /*
+    * Stage: Check that the pipeline isn't in inprogress or queued up. Once clear, run the regeneration job
+    */
+    context.stage("Check $javaVersion pipeline status") {
+
+      // Get all pipelines
+      def getPipelines = queryJenkinsAPI("https://ci.adoptopenjdk.net/job/build-scripts/api/json?tree=jobs[name]&pretty=true&depth1")
+
+      // Parse api response to only extract the relevant pipeline
+      getPipelines.jobs.name.each{ pipeline -> 
+        if (pipeline.contains("pipeline") && pipeline.contains(javaVersion)) {
+          Integer sleepTime = 900
+          Boolean inProgress = true
+
+          while (inProgress) {
+            // Check if pipeline is in progress using api
+            context.println "[INFO] Checking if ${pipeline} is running..." //i.e. openjdk8-pipeline
+
+            def pipelineInProgress = queryJenkinsAPI("https://ci.adoptopenjdk.net/job/build-scripts/job/${pipeline}/lastBuild/api/json?pretty=true&depth1")
+            inProgress = pipelineInProgress.building as Boolean
+
+            if (inProgress) {
+              // Sleep for a bit, then check again...
+              context.println "[INFO] ${pipeline} is running. Sleeping for ${sleepTime} seconds while waiting for ${pipeline} to complete..."
+              context.sleep sleepTime
+            }
+
+          }
+          
+          context.println "[SUCCESS] ${pipeline} is idle. Running regeneration job..."
+        }
+
+      }
+
+    } // end stage
+
+    /*
+    * Stage: Regenerate all of the job configurations by job type (i.e. jdk8u-linux-x64-hotspot
+    * jdk8u-linux-x64-openj9, etc.)
+    */
+    context.stage("Regenerate $javaVersion pipeline jobs") {
+
+      // Get downstream job folder and platforms
+      Map<String,List> downstreamJobs = [:]
+
+      // i.e. jdk11u, jdk8u, etc.
+      context.println "[INFO] Pulling downstream folders and jobs from API..."
+      def folders = queryJenkinsAPI("https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/api/json?tree=jobs[name]&pretty=true&depth=1")
+
+      folders.jobs.name.each { folder -> 
+        if ((folder == "jdk" && javaVersion == "jdk15") || folder.contains(javaVersion)) {
```

Comment:
Not sure there is a jdk15 job at the moment, but it will probably get called jdk15u

Reply:
`javaVersion` here is just being used to confirm which pipelines are being checked if they're running and also to identify which folders need to be regenerated downstream. The u doesn't matter until the JDK head changes (in which case, this line would need to be updated)